### ISSUE TITLE
[RFC] cookbook spring cleaning

### ISSUE THREAD

**Issue by c727** [home-assistant/home-assistant.github.io#4710]
Hi @SilvrrGIT @oakbrad @ntalekt @mf-social @mertenats @jjmontesl @happyleavesaoc @gstevenson @geekofweek @cy1701 @chriskacerguis @cbulock @bah2830 @aneisch @Vasiley @Teagan42 @stanvx @shortbloke @scottocs11 @skalavala @Haynie-Research-and-Development @Instagraeme @GreenTurtwig @Danielhiversen @danichispa @CiquattroFPV @CCOSTAN @arsaboo

you linked your HA config in our [docs/cookbook](https://home-assistant.io/cookbook/).
We want to do some spring cleaning... and as the config-repo is for some people something like the holy grail, I want to inform you about some upcoming changes:

Some configs/repos are outdated (breaking changes) or link to repos without any guidance - those will be removed.

You should think about why a user should check your config. Here are some notes to think about:
- If a user wants to see random config files he can search for `home-assistant-configuration` on GitHub and sort by most stars.
- You should keep your code in English or/and add comments
- If you want to link to your whole config, instead to a specific automation for example, your repo should have a README.md with summaries and links to specific topics or highlights.
- Users (.. who read cookbook) don't visit your repo for ref links to all your devices or a UI picture with 100 lamps, they want to see fancy automations or other cool integrations. So focus on them.
- If 3/4 of your panel is torrent related and all your automations are based around this, your repo is maybe inappropriate for the this.
- You should avoid installation instruction for HA in your linked files - some are outdated. Users reading cookbook have HA already installed. You can link to our official install docs for example and keep those updated ;) However, if you have some cool allround setup based on Hass.io add-ons or docker, you don't have to remove this!

I don't want you to censor anything in your repo or remove content, just make sure an interested user will find some useful info in your repo that he is looking for.

Here is a template for everyone who still wants to be listed in cookbook. It should help users to get a short summary/overview of the linked files (this is for the cookbook and not for the README.md in your repo):

```
**Name:**

your name or the "name" of your repo/config/automation/...


**Url:**

GitHub prefed, don't upload your whole config to the docs anymore


**Description:**

1-2 sentences that describe your repo


**Highlights (if you link to a whole repo):**

my diy-project XXX integrated in HA, advanced automations/ templates, python script to do XXX, automated glasshouse, voice automation flows to control the TV

```

!!! Fill out the template and submit it here: #4709
!!! This issue is for discussion !!!

Deadline: next week, if you miss the deadline you can submit your repo later again via PR :)

**Comment by mf-social** [comment]
I think this is a great idea. There's a fair few repos on there that I used when I was getting started, but haven't been updated in months so are well behind breaking changes.

I update mine several times a week, but can I suggest this... 

Anybody with a 'full config' repo includes their `.HAVERSION` file when uploading so you can easily see what version the config is valid on.

We have a limit as to how old the configuration is allowed to be. Maybe 2 months. This can be 'soft monitored' later. 

Anybody with a 'single file' inclusion puts a note at the top with the version it's valid for and a link to the blog for future-proofing breaking changes.

**Comment by c727** [comment]
2 months is a bit short but repos that are not updated in 1-2 years should be removed, if they focus in something broken for example

**Comment by mf-social** [comment]
There's a lot of breaking changes in 2 years. 🤔

**Comment by CCOSTAN** [comment]
More than a year is WAY too much. It needs to be on the order of months.. Maybe 6 as a happy medium.

**Comment by c727** [comment]
Keep in mind that users should not git clone your files, they should get some inspiration. However I'm open for discussion

**Comment by mf-social** [comment]
^ exactly.

**Comment by CCOSTAN** [comment]
users **_should_** not == They most **_definitely will_**. :)  lol

**Comment by geekofweek** [comment]
I'm not sure I 100% agree with frowning on listing out components used.  One of the primary reasons I started doing it was because I was getting a lot of feedback and requests on what I was using to do XYZ.  I was looking up and sending links often enough that I decided to start listing them.  I've also had a lot of comments on it that people actually like being able to see what it is I use to accomplish a lot of those automations.  Example is a lot of my occupancy automations use the Ecobee sensors to do what they do, but just seeing a sensor in the config isn't exactly explaining it.  Was easier than trying to do a write up for each and every automation I had a question on, I could point to the readme and say it was XYZ component that I used.  I'm open to removing the links that was also for the same reasons, people wanted to know where to get components so I just pointed them to Amazon of exactly what it was.  They're just standard links, I don't use a referral account or anything.

Same with the screen shots, I've had feedback (maybe not as much as above) that people want to see what the end state looks like.  Some newer users have expressed appreciation that they get an idea of what to expect in the end, etc.  

I could do a better job doing write up on each section, or specific automations.  I'll try to focus some time on that.  In the beginning it was just out of people asking for examples that I started out, and I actively point to specific bits on different forums online to show something.  Overall I like the idea of cleanup.

**Comment by c727** [comment]
I don't say remove your (ref) links or screenshots, ofc many like this. Just make sure your readme guides a bit through the "highlights" of your config. Only listing your devices and some screenshots is no guide (I did those notes without reference to any specific repo btw)

**Comment by geekofweek** [comment]
I hear you, just wanted to open some discussion.  I'm happy to take recommendations, criticism, etc.  I'm glad to see some open dialog about this.

**Comment by chriskacerguis** [comment]
Perhaps creating a simple bot that will run at some interval (every 3 months).  If there has not been any repo updates, then post an issue to the repo letting them know they have been flagged as "stale" and post why "You have not updated this repo in 3 months, in order to keep things fresh for new HA users you site will be delisted in 3 months unless..."  Give a 1 month "grace period", then after the month auto-remove it from the HA site, and post another issue staying it was delisted for being stale.

Some other thoughts:

> You should keep your code in English or/and add comments

This seems...unnecessary, and exclusionary. 

> Users (.. who read cookbook) don't visit your repo for ref links to all your devices or a UI picture with 100 lamps, they want to see fancy automations or other cool integrations.

I like looking to see what hardware people have, and links make it easier to find them.  As far as UI pics, etc.  I think that beauty is in the eye of the beholder.  Sometimes I take inspiration from what I see (hardware list, pics, etc).


I think that people should be able to post what they want on their repo, and share it on the site.  The only requirement should be that it is not "stale" or "abandoned" code.  I would think that this could be fixed with a bot (see above).

Just my thoughts.

**Comment by skalavala** [comment]
I think we should change the default search to **most recently updated** rather than based on **stars**. In my opinion, starts won't have any meaning if it is not updated, whereas most recently updated means we have a better chance of finding config that actually works.

**Comment by arsaboo** [comment]
Most recently updated **alone** is not entirely useful.

**Comment by skalavala** [comment]
I agree @arsaboo I also put some of the high starred repositories in the same category 😄

**Comment by CCOSTAN** [comment]
Which ones @skalavala?  Maybe we can find some obscure criteria to remove them...

**Comment by mf-social** [comment]
Ben's got two repos in the top 10 for stars, and both are over a year old now.

Shame, because it was his videos that brought a lot of people to the platform.

But suffice to say they're both 'super' outdated.

(see what I did there 😜)

**Comment by stanvx** [comment]
It would be ideal to be able to filter the search only showing those which have been updated in the last 3 - 4 months and then order by stars.

**Comment by mf-social** [comment]
Yeah, if such a thing is possible that would get my vote (if we were voting).

**Comment by Instagraeme** [comment]
I don't really understand the 'expiry' argument. Just because I made an interesting automation in 0.21 that uses fairly vanilla code doesn't mean that it's out of date and not useful to someone now - unless the changes to how home assistant is configured are going to be dramatically different in the future. Ultimately I would suggest the cookbook is most useful to beginners and if it is full of fantastic but esoteric configurations I think the point is lost.

Also, if we just look at most recently updated that doesn't necessarily mean they are the best examples. In fact they could be awful but would appear to be the 'top' result. I fully agree stars aren't the way forward since that will probably promote the ones that have been around longest.

As far as listing of components - I think this is an essential part. If I can see that someone is using a Acme Bulb and I have an Acme bulb then I'm probably interested in their config to see what they are doing with theirs. This is even more important when custom components are being used.

Ultimately I think the only real solution is to have a human curated section. That way the best configurations should be showcased. What the criteria for best configuration is - well that's a different discussion.

However, I would suggest that it's not the list of configurations that need to be worked on so much as the automation examples. These are too product specific and in all honesty a bit dull. I found configurations more useful than the examples when I was learning but, in my opinion, a really good set of examples, exploring different scenarios, is what was missing for me.
